base/tests/types.rs

```diff
@@ -50,62 +73,95 @@ fn show_record() {
+    let f = Type::function(vec![data("a", vec![])], Type::string());
+    let test = data("Test", vec![data("a", vec![])]);
+    let typ = Type::record(vec![Field {
+                                    name: "Test",
+                                    typ: Alias::new("Test",
+                                                    vec![Generic {
+                                                             kind: Kind::typ(),
+                                                             id: "a",
+                                                         }],
+                                                    f.clone()),
                                 }],
                            vec![Field {
                                     name: "x",
                                     typ: Type::int(),
                                 },
                                 Field {
                                     name: "test",
+                                    typ: Type::function(vec![data("Test",
+                                                                  vec![Type::int(), f.clone()]),
+                                                             Type::float(),
+                                                             f.clone(),
+                                                             f.clone()],
+                                                        f.clone()),
+                                },
+                                Field {
+                                    name: "record_looooooooooooooooooooooooooooooooooong",
+                                    typ: some_record(),
+                                },
+                                Field {
+                                    name: "looooooooooooooooooooooooooooooooooong_field",
                                     typ: test.clone(),
                                 }]);
-    assert_eq!(format!("{}", typ),
-               "{ Test a = a -> String, x: Int, test: Test a }");
-    let typ = Type::record(vec![Field {
-                                    name: "Test",
-                                    typ: Alias::new(
-                                        "Test",
-                                        vec![Generic {
-                                                       kind: Kind::typ(),
-                                                       id: "a",
-                                                   }],
-                                        f.clone(),
-                                    ),
-                                }],
-                           vec![]);
-    assert_eq!(format!("{}", typ), "{ Test a = a -> String }");
+    let expected = r#"{
+    Test a = a -> String,
+    x: Int,
+    test: Test Int (a -> String) ->
+        Float ->
+        (a -> String) ->
+        (a -> String) ->
+        a ->
+        String,
+    record_looooooooooooooooooooooooooooooooooong: {
+        Test a = a -> String,
```

Comment:
It might not actually be possible to write that out. The only way to 'export' a type that I know are used is
```f#
type Test = Float
{ Test }
```
but actually writing the *type declaration* may not be possible